### PR TITLE
feat: custom role for otomi api 

### PR DESCRIPTION
**Pull request by j-zimnowoda**
This change ensures that otomi-api has always right to read secrets and services.
Further improvement may be made to narrow down the scope of accessed secrets. However it is difficult to predict secret names and namespaces.

## Checklist

- [ ] Architecture Design Records have been added as `adr/*.md` and appended to list in `adr/_index.md`, if applicable.
- [ ] The `values-schema.yaml` file and `test/**` fixtures have been updated to reflect code changes, if applicable.
- [ ] The OpenApi Schema from redkubes/otomi-api project is compatible with definitions from `values-schema.yaml` file, if applicable.
- [ ] Helm releases are meeting otomi's baseline security policies, if applicable.
- [ ] Helm chart and helmfile changes are tested against upgrade scenario, if applicable.

### CODE REVIEW

**Review by Morriz — COMMENTED** [review]
Just a question from a security perspective: what are the use cases that api is given `secret` and `service` read/list access?

File: charts/otomi-api/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+metadata:
+  name: {{ template "otomi-api.serviceAccountName" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  {{- with .Values.serviceAccount.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    {{- include "otomi-api.labels" . | nindent 4 }}
+    {{- with .Values.serviceAccount.labels }}
```
Comment (Morriz) [inline charts/otomi-api/templates/serviceaccount.yaml:19]: question: don't these need to also exist in the chart's values? I think helm 3 uses the values file as a sort of schema to determine what is expected, and throws an error if other stuff is provided. I am referring to these missing values:
- `.Values.serviceAccount.annotations`
- `.Values.serviceAccount.labels`

**Comment by j-zimnowoda** [comment]
> Just a question from a security perspective: what are the use cases that api is given `secret` and `service` read/list access?

list services because we want to render available service in otomi-console for user convenience.
list of secrets, hmm now I am thinking that just `get` could be enough, because we know secret names we want to access.

File: charts/otomi-api/templates/serviceaccount.yaml (hunk 1 above)
Comment (j-zimnowoda) [reply charts/otomi-api/templates/serviceaccount.yaml:19]: helm3 does not throw error, however I can add those to values.

**Pull request merged**